packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -534,75 +599,601 @@ export const LatestDatePrimitiveInstanceValueEditor: React.FC = () => (
+          Unit
+        </div>
+        <CustomSelectorInput
+          className="query-builder-value-spec-editor__date-option-dropdown__custom-date__input-dropdown"
+          options={Object.values(QUERY_BUILDER_DATE_DURATION_UNIT_LABEL).map(
+            (t) => ({ value: t.toString(), label: t.toString() }),
+          )}
+          onChange={(durationUnit: CustomDateInputOption): void => {
+            setDurationUnitValue(durationUnit.value);
+            updateValue(
+              durationRangeValue,
+              durationUnit.value,
+              adjustmentValue,
+              startDateValue,
+            );
+          }}
+          value={{ value: durationUnitValue, label: durationUnitValue }}
+          darkMode={true}
+        />
+      </div>
+      <div className="query-builder-value-spec-editor__date-option-dropdown__custom-date__input">
+        <div className="query-builder-value-spec-editor__date-option-dropdown__custom-date__input-label">
+          Before/After
+        </div>
+        <CustomSelectorInput
+          className="query-builder-value-spec-editor__date-option-dropdown__custom-date__input-dropdown"
+          options={Object.values(QUERY_BUILDER_DATE_ADJUSTMENT).map((t) => ({
+            value: t.toString(),
+            label: t.toString(),
+          }))}
+          onChange={(adjustment: CustomDateInputOption): void => {
+            setAdjustmentValue(adjustment.value);
+            updateValue(
+              durationRangeValue,
+              durationUnitValue,
+              adjustment.value,
+              startDateValue,
+            );
+          }}
+          value={{ value: adjustmentValue, label: adjustmentValue }}
+          darkMode={true}
+        />
+      </div>
+      <div className="query-builder-value-spec-editor__date-option-dropdown__custom-date__input">
+        <div className="query-builder-value-spec-editor__date-option-dropdown__custom-date__input-label">
+          When
+        </div>
+        <CustomSelectorInput
+          className="query-builder-value-spec-editor__date-option-dropdown__custom-date__input-dropdown"
+          options={Object.values(
+            QUERY_BUILDER_CUSTOM_DATE_START_DATE_OPTION,
+          ).map((t) => ({
+            value: t.toString(),
+            label: t.toString(),
+          }))}
+          onChange={(startDate: CustomDateInputOption): void => {
+            setStartDateValue(startDate.value);
+            updateValue(
+              durationRangeValue,
+              durationUnitValue,
+              adjustmentValue,
+              startDate.value,
+            );
+          }}
+          value={{ value: startDateValue, label: startDateValue }}
+          darkMode={true}
+        />
+      </div>
+    </div>
+  );
+};
+
+export const CustomStartDayOfDateValueSpecificationEditor: React.FC<{
+  graph: PureModel;
+  customDateAdjustOptionValue: CustomDateOption;
```

Comment:
let's reorder `customDateAdjustOptionValue` to be the first param. Sorry for this nitpickiness :pray: